resteasy-core/src/main/java/org/jboss/resteasy/plugins/server/servlet/ServletContainerDispatcher.java

```diff
@@ -201,6 +201,11 @@ public void service(String httpMethod, HttpServletRequest request, HttpServletRe
             try {
                 headers = ServletUtil.extractHttpHeaders(request);
                 uriInfo = ServletUtil.extractUriInfo(request, servletMappingPrefix);
+            } catch (IllegalArgumentException e) {
+                response.sendError(HttpServletResponse.SC_NOT_ACCEPTABLE);
+                // made it warn so that people can filter this.
+                LogMessages.LOGGER.failedToParseRequest(e);
+                return;
```

Comment:
We can't do this here as there could be other reasons an `IllegalArgumentException` is thrown.

Reply:
Thank you very much for the feedback. I will close the PR and the issue as Obsolete.

Reply:
Thank you for all the research on this @khermano!

Reply:
You're welcome. Thank you for all your help and information.